torch/csrc/jit/runtime/static/passes.cpp

```diff
@@ -950,6 +921,113 @@ void FuseListUnpack(std::shared_ptr<torch::jit::Graph>& graph) {
   }
 } // namespace jit
 
+
+// Similar to FuseListUnpack but the operators results are being accessed by get_item rather than ListUnpack
+void FuseGetItemUnpack(std::shared_ptr<torch::jit::Graph>& graph, const c10::FastMap<c10::Symbol, c10::Symbol> unfused_to_fused) {
+  // replacement contains (old_node, new_node, list_unpack_node)
+  std::vector<std::tuple<Node*, c10::Symbol, std::vector<std::tuple<Node*, ArrayRef<Value*>>>>> replacement;
+  DepthFirstGraphNodeIterator graph_it(graph);
+  const auto get_item = fromQualString("aten::__getitem__");
+  for (auto node = graph_it.next(); node != nullptr; node = graph_it.next()) {
+    auto unfused_to_fused_it = unfused_to_fused.find(node->kind());
+    if (unfused_to_fused_it == unfused_to_fused.end()) {
+      continue;
+    }
+
+    const Value* value_out = node->outputs()[0];
+
+    const auto& new_sym = unfused_to_fused_it->second;
+
+    bool should_fuse = true;
+
+    std::vector<std::tuple<Node*, ArrayRef<Value*>>> user_to_remove;
+
+    for(auto index = 0; index < value_out->uses().size(); index++) {
+      Node* user = value_out->uses()[index].user;
+      if (user->kind() != get_item)
+      {
+        should_fuse = false;
+        break;
+      }
+      auto user_outputs = user->outputs();
+
+      if (user_outputs.empty()) {
+        should_fuse = false;
+        break;
+      }
+
+      if (shouldNotFuseListUnpackSpecialCase(node)) {
+        should_fuse = false;
+        break;
+      }
+      user_to_remove.emplace_back(user, user_outputs);
+    }
+
+
+    if (!should_fuse) {
+      continue;
+    }
+
+    replacement.emplace_back(node, new_sym, user_to_remove);
+  }
+
+  for(const auto& nodes : replacement) {
+    auto* old_node = std::get<0>(nodes);
+    auto new_sym = std::get<1>(nodes);
+    auto* new_node = graph->create(new_sym, 0);
+    for (Value* in : old_node->inputs()) {
+        new_node->addInput(in);
+      }
+    auto user_to_remove_nodes = std::get<2>(nodes);
+    for(const auto& nodes : user_to_remove_nodes) {
+      auto* user = std::get<0>(nodes);
+      auto user_outputs = std::get<1>(nodes);
+      for (Value* out : user_outputs) {
+        Value* new_out = new_node->addOutput();
+        new_out->copyMetadata(out);
+        out->replaceAllUsesWith(new_out);
+      }
+      user->destroy();
+    }
+    new_node->insertAfter(old_node);
+    old_node->destroy();
+  }
+} // namespace jit
+
+void FuseUnpack(std::shared_ptr<torch::jit::Graph>& graph){
+  const c10::FastMap<c10::Symbol, c10::Symbol> unfused_to_fused = {
```

Comment:
this map should probably be static const

Reply:
fixed